controllers/druid/drain_historical.go

```diff
@@ -0,0 +1,191 @@
+package druid
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	//TODO cleanup imports
+
+	"github.com/datainfrahq/druid-operator/apis/druid/v1alpha1"
+	appsv1 "k8s.io/api/apps/v1"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+var logger_drain_historical = logf.Log.WithName("drain_historical")
+
+type CoordinatorConfig struct {
+	decommissioningNodes []string
+}
+
+func checkDrainStatus(d DruidClient, start int32, end int32, podNames []string) (bool, error) {
+	historicalUsageStats, err := d.GetHistoricalUsage()
+	if err != nil {
+		return false, err
+	}
+	for it := start; it <= end; it++ {
+		if historicalUsageStats[podNames[it]] != 0 {
```

Comment:
If the pod name doesn't exist in the map of historicalUsageStats, then it would result in a 0 value. You might need to add a check to see if the pod name is in the map first.